pkg/trace/config/apply.go

```diff
@@ -466,7 +506,7 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	if config.Datadog.GetBool("apm_config.internal_profiling.enabled") {
 		profilingSite := config.Datadog.GetString("internal_profiling.profile_dd_url")
 		if profilingSite == "" {
-			profilingSite = site
+			profilingSite = config.Datadog.GetString("site")
```

Comment:
This change seems out of scope, or for another PR. I don't think we have a bug here, or do we?

Reply:
This is because of the refactoring. The site variable was defined earlier. But it got replaced with config.GetMainEndpoint.